src/Device.vala

```diff
@@ -85,6 +104,15 @@ public class Sound.Device : GLib.Object {
             default:
                 return input? _("Input") : _("Output");
         }
-    
+    }
+
+    public string? get_matching_profile (Device other_device) {
+        foreach (var profile in profiles) {
+            if (other_device.profiles.contains (profile)) {
```

Comment:
My only concern is if this function should return a single profile as it is comparing intersecting profiles between devices.

Reply:
I've discussed this a little bit with @naaando, the question is why would the first profile that intersects (exists for both input and output device) be the right profile? 
While I'm no expert the profile lists are [sorted on priority](https://github.com/elementary/switchboard-plug-sound/blob/e17be2b19ea2d0405f7b483160c3fa966f80c8ae/src/PulseAudioManager.vala#L619-L638) and this is what worked well for me in all my tests.